FirebaseAdmin/FirebaseAdmin/Messaging/LightSettings.cs

```diff
@@ -0,0 +1,194 @@
+        /// </summary>
+        [JsonIgnore]
+        public string Color
+        {
+            get
+            {
+                return this.lightColor.ColorString();
+            }
+
+            set
+            {
+                this.lightColor = LightSettingsColor.FromString(value);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the light on duration in milliseconds.
+        /// </summary>
+        [JsonIgnore]
+        public long LightOnDurationMillis { get; set; }
+
+        /// <summary>
+        /// Gets or sets the light off duration in milliseconds.
+        /// </summary>
+        [JsonIgnore]
+        public long LightOffDurationMillis { get; set; }
+
+        /// <summary>
+        /// Gets or sets the light settings color representation as accepted by the FCM backend service.
+        /// </summary>
+        [JsonProperty("color")]
+        private LightSettingsColor LightColor
+        {
+            get
+            {
+                return this.lightColor;
+            }
+
+            set
+            {
+                this.lightColor = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the string representation of <see cref="LightOnDurationMillis"/>.
+        /// </summary>
+        [JsonProperty("light_on_duration")]
+        private string LightOnDurationMillisString
+        {
+            get
+            {
+                return TimeConverter.LongMillisToString(this.LightOnDurationMillis);
+            }
+
+            set
+            {
+                this.LightOnDurationMillis = TimeConverter.StringToLongMillis(value);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the string representation of <see cref="LightOffDurationMillis"/>.
+        /// </summary>
+        [JsonProperty("light_off_duration")]
+        private string LightOffDurationMillisString
+        {
+            get
+            {
+                return TimeConverter.LongMillisToString(this.LightOffDurationMillis);
+            }
+
+            set
+            {
+                this.LightOffDurationMillis = TimeConverter.StringToLongMillis(value);
+            }
+        }
+
+        /// <summary>
+        /// Copies this Light Settings, and validates the content of it to ensure that it can be
+        /// serialized into the JSON format expected by the FCM service.
+        /// </summary>
+        internal LightSettings CopyAndValidate()
+        {
+            // Copy and validate the leaf-level properties
+            var copy = new LightSettings()
+            {
+                Color = this.Color,
+                LightOnDurationMillis = this.LightOnDurationMillis,
+                LightOffDurationMillis = this.LightOffDurationMillis,
+            };
+
+            return copy;
+        }
+
+        /// <summary>
+        /// The LightSettings Color object as expected by the FCM backend service.
```

Comment:
Suggest `LightSettingsColor` object

(Does this respect markdown backticks?  Otherwise, <code> tags.)

Reply:
Thanks, updated the comments.